swig/cpp/src/Connection.cpp

```diff
@@ -0,0 +1,42 @@
+#include <iostream>
+#include <stdexcept>
+
+#include "Connection.h"
+
+extern "C" {
+#include "sysrepo.h"
+}
+
+using namespace std;
+
+Connection::Connection(const char *app_name, const sr_conn_options_t opts)
+{
+    int rc;
+    _app_name = (char *) app_name;
```

Comment:
There is no need to store app_name (it only needs to be passed to sr_connect). Anyway, if you want to store it for any reason, you should duplicate the string, since it may get invalid though the lifetime of the Connection.